### PR TITLE
Remove removed ruff rule TRY200

### PR DESCRIPTION
**Pull request by ringohoffman**
My TOML linter is complaining that "TRY200" is not acceptable for the `tool.ruff.lint` schema.

From the ruff docs: https://docs.astral.sh/ruff/rules/reraise-no-cause/

> This rule has been removed and its documentation is only available for historical reasons.
> 
> This rule is identical to [B904](https://docs.astral.sh/ruff/rules/raise-without-from-inside-except/) which should be used instead.

and we are currently explicitly ignoring B904.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -51,7 +51,7 @@ ignore = [
     "B019",
     "B023",
     "B028", # No explicit `stacklevel` keyword argument found
-    "B904", # Migrate from TRY200
+    "B904", # Re-raised error without specifying the cause via the from keyword
```
Comment (Skylion007) [inline pyproject.toml:54]: Leaving it in the ignore list just disables. We need to explicitly select it, not ingore it.

Comment (ringohoffman) [reply pyproject.toml]: I didn't see they were in different lists. Thanks!

**Comment by Skylion007** [comment]
Pretty sure all the noqas will need to be adjusted too, or the original errors will need to be fixed.

**Review by Skylion007 — APPROVED** [review]

**Comment by ringohoffman** [comment]
@Skylion007 I guess `B904` is somehow stricter than `TRY200`?

Any recommendation on these: https://github.com/pytorch/pytorch/actions/runs/9100510101/job/25019431692#step:11:243

File: torch/_dynamo/variables/sdpa.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ def var_getattr(self, tx, name: str) -> VariableTracker:
             getattr_static(torch._C._SDPAParams, name)
         except AttributeError:
             # Using raise from is too verbose here
-            raise Unsupported(  # noqa: TRY200
+            raise Unsupported(  # noqa: B904
```
Comment (Skylion007) [inline torch/_dynamo/variables/sdpa.py:68]: Instead noqa here, do a `raise ... from None` to show that it's explicit according to the comment.

**Pull request closed without merging**